### ISSUE TITLE
Fix Lint/UselessMethodDefinition warning when AllowComments is disabled

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#10517]
With the upgrade to Rubocop 1.27.0, I'm seeing this warning appear multiple times when running RuboCop. I believe this is happening due to this [recent change](https://github.com/rubocop/rubocop/pull/10491) but not sure. I can conform _this does not happen_ with the previous version of 1.26.1.

## Expected behavior

I would expect to not see any warnings to appear when using this YAML configuration:

``` 
Lint/UselessMethodDefinition:
  Enabled: true
  AllowComments: false
```

## Actual behavior

``` 
Warning: Lint/UselessMethodDefinition does not support AllowComments parameter.

Supported parameters are:

  - Enabled
```

Here are the full debug details if it helps as well:

<details>
  <summary>Debug Details</summary>

``` 
For /Users/bkuhlmann/Engineering/OSS/caliber: configuration from /Users/bkuhlmann/Engineering/OSS/caliber/.rubocop.yml
Inheriting configuration from /Users/bkuhlmann/Engineering/OSS/caliber/config/all.yml
Inheriting configuration from /Users/bkuhlmann/Engineering/OSS/caliber/config/ruby.yml
Default configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-1.27.0/config/default.yml
Warning: Lint/UselessMethodDefinition does not support AllowComments parameter.

Supported parameters are:

  - Enabled

Inheriting configuration from /Users/bkuhlmann/Engineering/OSS/caliber/config/rake.yml
configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-rake-0.6.0/config/default.yml
Inheriting configuration from /Users/bkuhlmann/Engineering/OSS/caliber/config/performance.yml
configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.13.3/config/default.yml
configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.13.3/config/default.yml
Inheriting configuration from /Users/bkuhlmann/Engineering/OSS/caliber/config/rspec.yml
configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.9.0/config/default.yml
configuration from /Users/bkuhlmann/.cache/frum/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.9.0/config/default.yml
Warning: Lint/UselessMethodDefinition does not support AllowComments parameter.

Supported parameters are:

  - Enabled

Warning: Lint/UselessMethodDefinition does not support AllowComments parameter.

Supported parameters are:

  - Enabled

Use parallel by default.
Running parallel inspection
Inspecting 11 files
Scanning /Users/bkuhlmann/Engineering/OSS/caliber/Gemfile
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/440ab7921a960f6d421c2b02bc1bfd37611a3c6b
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/Guardfile
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/6bfb3da44ddf60bdb69fb0a10f5855c28e2d36b3
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/Rakefile
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/3ed194191dbc962497e0d4b5e241c12a2f8f6a9b
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/bin/console
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/91b8a2b149f09611367d8f96a39f426133c3dc75
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/bin/guard
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/7b08bfd1bbdb7ff3db3dc89b305e0723dedf37d3
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/bin/rubocop
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/63d8f019a8cd27ac14fdf0b863018efd4d45c725
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/caliber.gemspec
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/e9943d1b666dbf9fabcd371b909b537ed851ce5b
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/lib/caliber.rb
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/3ad50d637b3ed2a2d0cebfbdb5bcf32ec02ccb51
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/spec/gemspec_spec.rb
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/4b2ce949df0aa584f4d352ce9dee9c83cfe550c8
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/spec/spec_helper.rb
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/438df7daf0d4e9607238a0cd61132ae3c4b5baf7
.Scanning /Users/bkuhlmann/Engineering/OSS/caliber/spec/support/shared_contexts/temp_dir.rb
Loading cache from /Users/bkuhlmann/.cache/rubocop_cache/b22ae3667085de063c415dec8417d49effe54b12/6d7a3b621ca1730e04accd938619e4bdab66cfb1/0941b3ac45183d27ddba68cd45f9cbd632e3f1f6
```
</details>

## Steps to reproduce the problem

If you use the YAML configuration mentioned, above, in the *Expected Behavior* section, you should see the warning immediately.

## RuboCop version

```
1.27.0 (using Parser 3.1.1.0, rubocop-ast 1.17.0, running on ruby 3.1.1 arm64-darwin21.4.0)
  - rubocop-performance 1.13.3
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.9.0
```

**Comment by nobuyo** [comment]
This behavior seems to be intentional since v1.27.0, changed at https://github.com/rubocop/rubocop/pull/10492.

**Comment by bkuhlmann** [comment]
Thanks. Hmm, so I guess I'm still confused. The [documentation](https://docs.rubocop.org/rubocop/cops_lint.html#lintuselessmethoddefinition) says you can use the `AllowComments` configuration. I guess this means `AllowComments` is no longer a configurable option?

**Comment by nobuyo** [comment]
Yes.

[This PR](https://github.com/rubocop/rubocop/pull/8696) fixes `Lint/UselessMethodDefinition` 's false negative.

My understanding is that `Lint/UselessMethodDefinition` just checks empty constructors and methods just delegating to `super`, so it doesn't matter if there are comments or not.
That is why the option was removed.

The source code in the documentation seems to be updated, so I think it just hasn't been published yet.

https://github.com/rubocop/rubocop/blob/master/docs/modules/ROOT/pages/cops_lint.adoc#lintuselessmethoddefinition

I'm sorry if I'm wrong.

**Comment by bkuhlmann** [comment]
OK, I didn't grok that properly, but I see what you are saying now. Thanks! :bow: 

I'm closing this issue since there is no further action and I can update on my side accordingly.
